histolab/mixins.py

```diff
@@ -0,0 +1,44 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------
+# Copyright 2021 All Histolab Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------
+
+import numpy as np
+
+
+class LinalgMixin:
+    @staticmethod
+    def normalize_columns(arr: np.ndarray) -> np.ndarray:
+        """Normalize each column vector in an array"""
+        return arr / np.linalg.norm(arr, axis=0)
+
+    @staticmethod
+    def two_principal_components(arr: np.ndarray) -> np.ndarray:
```

Comment:
<img width="794" alt="image" src="https://user-images.githubusercontent.com/31658006/163681376-06830555-7d76-45b1-bfa7-43135acdf772.png">
<img width="786" alt="image" src="https://user-images.githubusercontent.com/31658006/163681393-c2cc9baf-26b4-4d50-ae71-e7bb33c0ba05.png">

Reply:
We will stick with the implementation using the covariance matrix (method 1) as already implemented. The svd approach (method 2) will be considered if needed